### PR TITLE
[BALDE-30] Add more core logs

### PR DESCRIPTION
**Pull request by 4jeR**

### CODE REVIEW

File: src/events/callback/handler.cpp [hunk 1]
```diff
@@ -11,6 +11,6 @@ namespace Bald {
     Handler::~Handler() {
         auto iter = std::find(m_TakenID.begin(), m_TakenID.end(), m_ID);
         m_TakenID.erase(iter);
+        CORE_LOG_INFO("[Handler] Successfully removed current ID");
```
Comment (Blinkuu) [inline src/events/callback/handler.cpp]: This is a destructor, I think something like "Successfully destroyed handler object" would make more sense.

File: src/events/callback/handler.cpp [hunk 2]
```diff
@@ -11,6 +11,6 @@ namespace Bald {
...
     }
-
```
Comment (Blinkuu) [inline src/events/callback/handler.cpp]: We always keep one new line before namespace ends. You can change this for now, but hopefully, we will set up a normalized code style for our project.

File: src/events/callback/handler.h [hunk 3]
```diff
@@ -85,8 +85,10 @@ namespace Bald {
             }
 
             if(!handled) {
-                CORE_LOG_WARN("[Handler] Could not create Handler object because maximum number of ID's was reached");
+                CORE_LOG_WARN("[Handler] Could not create Handler object because maximum number of ID's was reached!");
```
Comment (Blinkuu) [inline src/events/callback/handler.h]: We either put full stops/commas/exclamation marks or we don't. From what I can see, until now we have not been using any punctuation after sentences.

Comment (Blinkuu) [reply src/events/callback/handler.h]: But you are free to pick one convention, but please check whether it is followed everywhere.

File: src/events/event_manager.cpp [hunk 4]
```diff
@@ -26,6 +26,7 @@ namespace Bald {
     }
 
     void EventManager::Flush(int n) noexcept {
+        CORE_LOG_INFO("[EventManager] About to begin flushing...");
```
Comment (Blinkuu) [inline src/events/event_manager.cpp]: I would change this to simply "Flushing events"

Comment (Blinkuu) [reply src/events/event_manager.cpp]: Or if we follow convention -> Add one more log after flush was finished successfully.

**Review by Blinkuu — CHANGES_REQUESTED** [review]

File: src/events/callback/handler.h [hunk 5]
```diff
@@ -87,6 +87,8 @@ namespace Bald {
             if(!handled) {
                 CORE_LOG_WARN("[Handler] Could not create Handler object because maximum number of ID's was reached");
             }
+            else
```
Comment (BIGbadEL) [inline src/events/callback/handler.h]: {} should be used in both cases or in neither of them

File: src/events/event_manager.h [hunk 6]
```diff
@@ -129,12 +129,13 @@ namespace Bald {
...
         }
-
+        CORE_LOG_INFO("[EventManager] Subscription completed successfully!");
         return m_Callbacks[typeid(T)]->back()->GetID();
     }
 
     template<class T>
-    void EventManager::Unsubscibe(unsigned id) noexcept {
+    void EventManager::Unsubscribe(unsigned id) noexcept {
+        CORE_LOG_INFO("[EventManager] About to begin unsubscribe...");
```
Comment (BIGbadEL) [inline src/events/event_manager.h]: add info about id

File: src/events/event_manager.h [hunk 7]
```diff
@@ -129,12 +129,13 @@ namespace Bald {
                 m_Callbacks[typeid(T)]->push_back(new AsyncFunctionHandler(callback, args...));
                 break;
         }
-
+        CORE_LOG_INFO("[EventManager] Subscription completed successfully!");
```
Comment (BIGbadEL) [inline src/events/event_manager.h]: add info about handler id

File: src/events/event_manager.h [hunk 8]
```diff
@@ -171,13 +175,15 @@ namespace Bald {
 
     template<class T>
     void EventManager::RemoveAllCallbacksByType() noexcept {
+        CORE_LOG_INFO("[EventManager] About to remove all callbacks by type...");
```
Comment (BIGbadEL) [inline src/events/event_manager.h]: add info about type name we should be able to decode it on Travis

**Review by BIGbadEL — CHANGES_REQUESTED** [review]

File: src/events/callback/handler.cpp (hunk 2 above)
Comment (piotrmoszkowicz) [reply src/events/callback/handler.cpp]: "hopefully" kappa

**Review by Blinkuu — APPROVED** [review]

**Review by BIGbadEL — APPROVED** [review]

**Pull request merged**